### PR TITLE
Fix Bug 1410378 - Add automated test for "Top Sites - Add Top Site"

### PR DESCRIPTION
**Pull request by vikasmahato**

### CODE REVIEW

**Comment by vikasmahato** [comment]
@Mardak Can you review this?

File: system-addon/test/functional/mochitest/browser_topsites_section.js [hunk 1]
```diff
@@ -54,3 +54,52 @@ test_newtab({
       "Topsite should be unpinned");
   }
 });
+
+// Check Topsites add
+test_newtab(
+  // it should be able to click the topsites edit button to reveal the edit topsites modal and overlay.
+  async function topsites_edit() {
```
Comment (Mardak) [inline system-addon/test/functional/mochitest/browser_topsites_section.js]: Let's name the test `topsites_add`

File: system-addon/test/functional/mochitest/browser_topsites_section.js [hunk 2]
```diff
@@ -54,3 +54,52 @@ test_newtab({
...
+    // Find the edit topsites button
+    const topsitesEditBtn = content.document.querySelector(".edit-topsites-button button");
```
Comment (Mardak) [inline system-addon/test/functional/mochitest/browser_topsites_section.js]: The "Edit" is gone and replaced with "Add" now that skips the edit modal, so the intermediate step of clicking "Add" after "Edit" is not necessary anymore either.

File: system-addon/test/functional/mochitest/browser_topsites_section.js [hunk 3]
```diff
@@ -54,3 +54,52 @@ test_newtab({
...
+    let addBtn = content.document.querySelector(".done.add");
+    addBtn.click();
+
+    // Click the "Done" button
+    let doneBtn = content.document.querySelector(".done");
+    doneBtn.click();
+
+    // Wait for Topsite to be populated
+    await ContentTaskUtils.waitForCondition(() => content.document.querySelector(".top-site-outer:nth-child(1) a"), "No Topsite found");
+    let topSite = content.document.querySelector(".top-site-outer:nth-child(1) a");
```
Comment (Mardak) [inline system-addon/test/functional/mochitest/browser_topsites_section.js]: Could probably just do `:first-child` and have the condition to be waited for just be that the top site's href is the expected value.

**Comment by vikasmahato** [comment]
I have updated the pull request

**Comment by piatra** [comment]
Doing another cross platform test run https://treeherder.mozilla.org/#/jobs?repo=try&revision=037d622c136b99aea72cf11f6bf0ccaca74bfa70.

**Comment by vikasmahato** [comment]
@piatra I have added the `waitForCondition` you requested

File: system-addon/test/functional/mochitest/browser_topsites_section.js [hunk 4]
```diff
@@ -59,3 +59,62 @@ test_newtab({
...
+    // Click the "Add" button
+    let addBtn = content.document.querySelector(".done");
+    addBtn.click();
+
+    // Wait for Topsite to be populated
+    await ContentTaskUtils.waitForCondition(() => (content.document.querySelector(".top-site-outer:first-child a").getAttribute("href") === "https://bugzilla.mozilla.org"), "No Topsite found");
+
+    // Remove topsite after test is complete
+    await ContentTaskUtils.waitForCondition(() => content.document.querySelector(".top-site-icon"),
+      "Topsite tippytop icon not found");
```
Comment (piatra) [inline system-addon/test/functional/mochitest/browser_topsites_section.js]: Hey. Everything looks good cross platform too! One last question: Why do this check? You already know from previous asserts that the topsite was added. It doesn't seem to reflect the comment above.

**Comment by vikasmahato** [comment]
@piatra Thanks for pointing it out. That condition check was indeed redundant. I have removed it and updated the pull request.

**Comment by as-pine-proxy** [comment]
[Treeherder: c4ee41881e22e154f6c593c9dd13b72c5f8cebb3](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=c4ee41881e22e154f6c593c9dd13b72c5f8cebb3)

**Pull request merged**